lib/array.js

```diff
@@ -108,7 +109,11 @@ internals.checkItems = function (items, wasArray, state, options) {
     var requireds = this._inner.requireds.slice();
     var inclusions = this._inner.inclusions.concat(requireds);
 
-    for (var v = 0, vl = items.length; v < vl; ++v) {
+    if (this._inner.ordered.length && !this._inner.items.length) {
```

Comment:
If `ordered()` has been called without `items()`, we would need to make sure it errors when there are more elements than `ordered` items. (Or do we not want to do that?)